### PR TITLE
New version: Waterfox 6.0.1

### PR DESCRIPTION
**Pull request by aaronliu0130**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/120925)

### CODE REVIEW

**Comment by aaronliu0130** [comment]
But it's the exact same installer as #120919 and /S on my machine works

**Comment by stephengillie** [comment]
Automatic Validation ended with:
image

File: manifests/w/Waterfox/Waterfox/G6.0.1/Waterfox.Waterfox.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+FileExtensions:
+- htm
+- html
+- pdf
+- url
+ProductCode: Waterfox 115.3.0 (x64 en-US)
+AppsAndFeaturesEntries:
+- DisplayName: Waterfox (x64 en-US)
+  DisplayVersion: 115.3.0
+  ProductCode: Waterfox 115.3.0 (x64 en-US)
```
Comment (stephengillie) [inline manifests/w/Waterfox/Waterfox/G6.0.1/Waterfox.Waterfox.installer.yaml:28]:
```suggestion
  ProductCode: Waterfox 115.3.0 (x64 en-US)
Dependencies:
  PackageDependencies:
  - PackageIdentifier: Microsoft.VCRedist.2015+.x64
```

**Comment by aaronliu0130** [comment]
Oh my, what other dependencies do we need? Is there a quick way of testing it myself?

**Comment by aaronliu0130** [comment]
On my windows sandbox, running (installer.exe) /S after installing VC_redist.x64 (2015+) works without a hitch. I didn't use winget to install it though as downloading in windows sandbox is excruciating slow, bing takes minutes to load

**Comment by stephengillie** [comment]
Automatic Validation ended with:
image

Even with the dependency preinstalled, it appears to ask. I'll manually validate to check.

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 5:08:34 PM:  > 2023-09-27 17:08:33.963 [CLI ] Package hash verification failed. SHA256 in manifest 
[2025ea03c48afad4582ab7b2d39f413aff2bf8b431923db9321ec1862ed124cd] does not match download 
[38a889a46b6210a15d94aeb1169366fd4c444b9b79e0f94e81a0da6f47e5da7a]

File: manifests/w/Waterfox/Waterfox/G6.0.1/Waterfox.Waterfox.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+Dependencies:
+  PackageDependencies:
+  - PackageIdentifier: Microsoft.VCRedist.2015+.x64
+Installers:
+- Architecture: x64
+  InstallerUrl: https://cdn1.waterfox.net/waterfox/releases/G6.0.1/WINNT_x86_64/Waterfox%20Setup%20G6.0.1.exe
+  InstallerSha256: 2025ea03c48afad4582ab7b2d39f413aff2bf8b431923db9321ec1862ed124cd
```
Comment (stephengillie) [inline manifests/w/Waterfox/Waterfox/G6.0.1/Waterfox.Waterfox.installer.yaml:35]:
```suggestion
  InstallerSha256: 38A889A46B6210A15D94AEB1169366FD4C444B9B79E0F94E81A0DA6F47E5DA7A
```

**Comment by aaronliu0130** [comment]
@stephengillie Also, do we still need the VCRedist dependency with the correct checksum? These errors sound unrelated though...

**Comment by stephengillie** [comment]
@aaronliu0130 Yes the dependency is still needed. The two green labels means it's passed validation and ready for a moderator to approve.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**